ipatests/pytest_ipa/integration/firewall.py

```diff
@@ -120,3 +221,57 @@ def remove_passthrough_rules(self, rules, ipv=None):
         """
         for rule in rules:
             self.passthrough_rule(["-D"] + rule, ipv)
+
+
+class Firewall(FirewallBase):
+    """
+    Depending on the ipaplatform proxy firewall tasks to the actual backend.
+    Current supported backends: firewalld and no-op firewall.
+    """
+    def __init__(self, host):
+        """Initialize with host where firewall changes should be applied"""
+        # break circular dependency
+        from .tasks import get_platform
+
+        self.host = host
+        platform = get_platform(host)
+
+        firewalls = {
+            'rhel': FirewallD,
+            'fedora': FirewallD,
+            'debian': FirewallD,
+            'ubuntu': FirewallD,
+            'altlinux': NoOpFirewall,
+        }
+        if platform not in firewalls:
+            raise ValueError(
+                "Platform {} hasn't supported Firewall".format(platform))
```

Comment:
nit: hasn't supported -> doesn't support

Reply:
Ohh, my perfect English :smile: . Thank you, @rcritten.